code/datums/components/hallucinations.dm

```diff
@@ -549,9 +600,138 @@ ABSTRACT_TYPE(/datum/component/hallucination)
+	desc = "Perhaps the densest thing in existence, except for you."
+	icon = null
+	icon_state = null
+	density = 0
+	var/list/my_image_overrides = list()
+	var/list/eaten_atoms = list()
+	var/mob/my_target = null
+	var/spaget_count = 0
+	var/right_spinning = 0
+
+/obj/fake_singulo/New(var/loc, var/mob/target, var/ttl = 60 SECONDS)
+	..()
+	SPAWN(ttl)
+		qdel(src)
+	src.right_spinning = prob(50)
+	src.my_target = target
+	var/image/client_image = image(icon = 'icons/effects/64x64.dmi', icon_state = "whole", loc = src)
+	client_image.override = TRUE
+	client_image.plane = PLANE_DEFAULT_NOWARP
+	var/image/lense = image(icon='icons/effects/overlays/lensing.dmi', icon_state="lensing_med_hole", pixel_x = -208, pixel_y = -208, loc = src)
+	lense.plane = PLANE_DISTORTION
+	lense.blend_mode = BLEND_OVERLAY
+	lense.appearance_flags = RESET_ALPHA | RESET_COLOR
+	src.my_image_overrides += client_image
+	src.my_image_overrides += lense
+	target << client_image
+	target << lense
+	process()
+
+/obj/fake_singulo/disposing()
+	if(src.my_image_overrides && my_target?.client)
+		for(var/client_image in src.my_image_overrides)
+			my_target.client?.images -= client_image
+			qdel(client_image)
+	qdel(src.my_image_overrides) //not really necessary, but why not
+	qdel(src.eaten_atoms)
+	. = ..()
 
 
+/obj/fake_singulo/proc/process()
+	var/target_dist = get_dist(src, src.my_target)
+	if(target_dist > 20) //if offscreen by a good amount
+		qdel(src)
+		return
+	else if(target_dist < 6)
+		//oh no you're being pulled into the singularity!
+		if(prob(50))
+			src.my_target.step_towards_movedelay(src) //respect standard movements, this isn't actual gravity, you're just stepping cos you're hallucinating
+
+	//otherwise, we're pretending to be a singularity
+	if(prob(30))
+		step(src, pick(cardinal))
+	else
+		step_towards(src, src.my_target) //oh god it's chasing me!
+
+	//"eat" stuff by overriding its icon with a blank icon state
+	for (var/X in range(3, src))
+		if (!X)
+			continue
+		if (X == src)
+			continue
+		var/atom/A = X
+
+		if (A.event_handler_flags & IMMUNE_SINGULARITY)
+			continue
+
+		if (A.event_handler_flags & IMMUNE_SINGULARITY_INACTIVE)
+			continue
+
+		if (!isarea(X))
+			if(IN_EUCLIDEAN_RANGE(src, A, 2.5))
+				if (X == src.my_target) //gotcha!
+					if(!ON_COOLDOWN(src.my_target, "fake_singulo_scream", 5 SECONDS))
+						src.my_target.emote("scream", FALSE)
+					src.my_target.changeStatus("weakened", 2 SECONDS)
+					src.my_target.changeStatus("stunned", 2 SECONDS)
+					SPAWN(5 SECONDS)
+						qdel(src)
+
+				if(A in src.eaten_atoms)
+					continue
+				//this is a thing we're going to eat, so blank client image override for you
+				var/image/blank_image = image(loc = A)
+				blank_image.override = TRUE
+				src.my_image_overrides += blank_image
+				src.eaten_atoms += X
+				src.my_target << blank_image
+				//for the spinny falling in animations. low count for performance
+				//this is almost a straight copy/paste from singulo code
```

Comment:
Could this be a animation library proc? It's likely to end up out of sync with the real singulo otherwise when people make changes